staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -261,11 +261,18 @@ func patchResource(
 			// 3. ensure no conflicts between the two patches
 			// 4. apply the #1 patch to the currentJS object
 
-			// Since the patch is applied on versioned objects, we need to convert the
-			// current object to versioned representation first.
-			currentVersionedObject, err := unsafeConvertor.ConvertToVersion(currentObject, kind.GroupVersion())
```

Comment:
currentObject should already be in the specified version for CRD, right? if so, this should be a no-op. If ConvertToVersion errors, I'd expect that to be fixed to recognize the object is already in the desired version and return

Reply:
The first one is, the conversion to internal is not.

Reply:
Of course we could make it into a noop and hope nothing else breaks.

Reply:
this works:

```diff
--- a/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go
+++ b/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go
@@ -832,6 +832,9 @@ func (UnstructuredObjectConverter) ConvertToVersion(in runtime.Object, target ru
                        // TODO: should this be a typed error?
                        return nil, fmt.Errorf("%v is unstructured and is not suitable for converting to %q", kind, target)
                }
+               if gvk.Version == runtime.APIVersionInternal {
+                       return in, nil
+               }
                in.GetObjectKind().SetGroupVersionKind(gvk)
        }
        return in, nil
```

Reply:
If conversion to an internal version is encountered, it treats it as a noop.

Reply:
Hmm, then the converter does not really what it is told to do.

Reply:
I played with https://github.com/kubernetes/kubernetes/pull/55001. But am not convinced it's worth the complexity. @liggitt's idea above is simpler.